altair/utils/data.py

```diff
@@ -85,18 +87,29 @@ def to_json(data, prefix='altair-data'):
     check_data_type(data)
     ext = '.json'
     filename = _compute_filename(prefix=prefix, ext=ext)
+    data_format = {'type': 'json'}
     if isinstance(data, pd.DataFrame):
         data = sanitize_dataframe(data)
-        data.to_json(filename, orient='records')
+        if not hasattr(data,'__geo_interface__'):
+            data.to_json(filename, orient='records')
+        else: #GeoPandas
+            with open(filename,'w') as f:
+                json.dump(data.__geo_interface__, f)
+            data_format['property']='features'
+
+    elif hasattr(data,'__geo_interface__'): # geojson object
+        with open(filename,'w') as f:
+            json.dump(data.__geo_interface__, f)
```

Comment:
Well, `GeoPandas` needs extra line of code than a simple `__geo_interface__`.  I had something like :
```python
if isinstance(data, pd.DataFrame) and hasattr(data, '__geo_interface__'):
    # do GeoPandas stuff
elif hasattr(data, '__geo_interface__'):
    # do geo_interface stuff
elif isinstance(data, pd.DataFrame):
    # do dataframe stuff
elif isinstance(data, dict):
   # do dict stuff
````
That is simple and clear. But I supposed that it could provoke errors in case some modification in Pandas block without mirroring it to GeoPandas. That why is so strange logic in final version.

Should i rewrite it back?

Reply:
Ah, I see – there's overlap between geo and dataframe.

I think it would be best to put all the geo logic under one if statement, so something like:
```
if hasattr(data, '__geo_interface__'):
    if dataframe:
        #stuff
    else:
        # other stuff
elif isinstance(data, pd.DataFrame):
    # do dataframe stuff
elif isinstance(data, dict):
   # do dict stuff
```

Reply:
Done